geozero/src/wkt/wkt_writer.rs

```diff
@@ -255,8 +273,18 @@ mod test {
     #[cfg(feature = "with-wkb")]
     fn from_wkb() {
         let blob = hex::decode("01040000A0E6100000020000000101000080000000000000244000000000000034C0000000000000594001010000800000000000000000000000000000E0BF0000000000405940").unwrap();
-        let mut cursor = std::io::Cursor::new(blob);
-        let ewkt = Ewkt::from_wkb(&mut cursor, WkbDialect::Ewkb).unwrap();
-        assert_eq!(ewkt.0, "SRID=4326;MULTIPOINT(10 -20 100,0 -0.5 101)")
+        let ewkt = Ewkb(blob)
+            .to_wkt_with_opts(WktDialect::Ewkt, CoordDimensions::xyz(), Some(4326))
+            .unwrap();
+        assert_eq!(ewkt, "SRID=4326;MULTIPOINT Z(10 -20 100,0 -0.5 101)")
```

Comment:
Similarly, the API just isn't sufficient... There's no way to get the coordinate dimension from the `EWKB` without consuming it. And so there's no way right now to know the SRID and CoordDimensions of the input and pass that on to the writer at the beginning.